File: codingTest/baekjoon/python/16165.py
# N, M = map(int, input().split())
# team = dict()
# while N != 0:
#     teamName = input()
#     team[teamName] = [input() for _ in range(int(input()))]
#     N -= 1
# while M != 0:
#     quiz, check = input(), int(input())
#     if check:
#         for k, v in team.items():
#             if quiz in v:
#                 print(k)
#                 break
#     else:
#         for i in sorted(team.get(quiz)):
#             print(i)
#     M -= 1

N, M = map(int, input().split())

teamMem, memTeam = {}, {}
for _ in range(N):
    teamName, memNum = input(), int(input())
    teamMem[teamName] = []
    for _ in range(memNum):
        name = input()
        teamMem[teamName].append(name)
        memTeam[name] = teamName

for i in range(M):
    quiz, check = input(), int(input())
    if check:
        print(memTeam[quiz])
    else:
        for mem in sorted(teamMem[quiz]):
            print(mem)

File: codingTest/baekjoon/javascript/16165.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on('line', line => {
  input.push(line);
}).on('close', () => {
  const TEAM_MEMBER = {};
  const MEMBER_TEAM = {};
  let [N, M] = input.shift().split(' ');
  while (N--) {
    let teamName = input.shift();
    let memberNumber = +input.shift();
    TEAM_MEMBER[teamName] = [];
    while (memberNumber--) {
      let memberName = input.shift();
      TEAM_MEMBER[teamName].push(memberName);
      MEMBER_TEAM[memberName] = teamName;
    }
  }

  while (M--) {
    let quiz = input.shift();
    let check = +input.shift();
    if (check) console.log(MEMBER_TEAM[quiz]);
    else for (i of TEAM_MEMBER[quiz].sort()) console.log(i);
  }
});
